extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -723,13 +724,13 @@ private void applyBuildItems(Session session,
         });
 
         kubernetesEnvs.forEach(e -> {
-            session.resources().decorate(e.getTarget(), new AddEnvVarDecorator(new EnvBuilder()
-                    .withName(EnvConverter.convertName(e.getName()))
-                    .withValue(e.getValue())
-                    .withSecret(e.getSecret())
-                    .withConfigmap(e.getConfigMap())
-                    .withField(e.getField())
-                    .build()));
+            String containerName = kubernetesName;
```

Comment:
Why do we have a different container name per deployment target?

Reply:
We don't really, but what would we use instead to determine the container name?

Reply:
I see, we are just covering the edge case where the user decided to use different names. I am covered then!